File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ret = [[1]]
        for i in range(1, numRows):
            ret.append([1] + [ret[i - 1][j - 1] + ret[i - 1][j] for j in range(1, i)] + [1])
        return ret

File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ret;
        for (int i = 1; i <= numRows; ++i) {
            vector<int> sub(i);
            for (int j = 1; j <= i; ++j) {
                if (j == 1 || j == i) sub[j - 1] = 1;
                else sub[j - 1] = ret.back()[j - 1] + ret.back()[j - 2];
            }
            ret.push_back(sub);
        }
        return ret;
    }
};